src/Statement.php

```diff
@@ -457,6 +458,28 @@ public function getClauses()
         return static::$CLAUSES;
     }
 
+    /**
+     * Gets the clause order of this statement as an array
+     * with clause as key and index as value.
+     *
+     * @return array<string, int>
+     */
+    public function getClauseOrder(): array
+    {
+        $clauses = [];
+        foreach (array_keys($this->getClauses()) as $key) {
+            if ($key === '_END_OPTIONS') {
+                if (static::$END_OPTIONS !== []) {
+                    array_push($clauses, ...array_keys(static::$END_OPTIONS));
+                }
```

Comment:
When mergin to master (PHP >= 7.3) the if is no long needed
```suggestion
                array_push($clauses, ...array_keys(static::$END_OPTIONS));
```
https://www.php.net/manual/en/function.array-push.php